v4-client-js/src/clients/native.ts

```diff
@@ -1352,3 +1352,92 @@ export async function setSelectedGasDenom(gasDenom: string): Promise<string> {
     return wrappedError(error);
   }
 }
+
+export async function getMegavaultOwnerShares(payload: string): Promise<string> {
+  try {
+    const client = globalThis.client;
+    if (client === undefined) {
+      throw new UserError('client is not connected. Call connectClient() first');
+    }
+    const json = JSON.parse(payload);
+    const address = json.address;
+    if (address === undefined) {
+      throw new UserError('address is not set');
+    }
+    const response =
+      await globalThis.client?.validatorClient.get.getMegavaultOwnerShares(address);
+    return encodeJson(response);
+  } catch (e) {
+    return wrappedError(e);
+  }
+}
+
+export async function getMegavaultWithdrawalInfo(
+  sharesToWithdraw: bigint
+): Promise<string> {
+  try {
+    const client = globalThis.client;
+    if (client === undefined) {
+      throw new UserError('client is not connected. Call connectClient() first');
+    }
+    const response =
+      await globalThis.client?.validatorClient.get.getMegavaultWithdrawalInfo(sharesToWithdraw);
+    return encodeJson(response);
+  } catch (e) {
+    return wrappedError(e);
+  }
+}
+
+export async function depositToMegavault(
+  subaccountNumber: number,
```

Comment:
I thought we need to decode the payload for params?

Reply:
We use the encoded payload when calling v4-client functions from Abacus.  In this case, we will call those functions from the mobile client directly instead of having Abacus calling it.

`getMegavaultOwnerShares` will be called from Abacus, so it's using payload string.

Reply:
make sense, thanks!